docs/decoding.md

```diff
@@ -360,3 +360,104 @@ And now, the Json decoder for Animal can handle both formats:
 """{"name": "Snake", "categories": [ "Cold-blooded", "Reptile"]}""".fromJson[Animal]
 // >>  Right(Animal(Snake,List(Cold-blooded, Reptile)))
 ```
+
+# JSON AST and Cursors
+
+In most cases it is not necessary to work with the JSON AST directly,
+instead it is more convenient to decode directly to domain objects.
+However, sometimes it is handy to work with a lower level representation of JSON.
+This may for example be the case when you need to work with deeply nested JSON structures 
+that would result in deeply nested case classes,
+or when you expect a lot of variation in the JSON structure, which would result in nasty decoders.
+
+
+## JSON AST
+
+To get the AST representation of a JSON string, use the `fromJson[Json]` method.
+
+```scala
```

Comment:
I prefer using compile-safe code snippets to keep the documents up-to-date using `mdoc` modifiers.

Reply:
right... it even says so in the docs contribution guidelines. Sorry, missed that - will update my PR.

Reply:
done.

Note that the CI now fails because of an non-up-tp-date README file, which seems to relate to https://github.com/zio/zio-json/commit/61b7634b1ea3d7062298c9ab8a4c5f996800fef0

Reply:
> Note that the CI now fails because of an non-up-tp-date README file, which seems to relate to [61b7634](https://github.com/zio/zio-json/commit/61b7634b1ea3d7062298c9ab8a4c5f996800fef0)

@BalduinLandolt 
Please update the PR from upstream `series/2.x` branch.

Reply:
@khajavi thanks for updating it! Should be good to merge now?